modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/util/MDMUtil.java

```diff
@@ -32,7 +32,7 @@ public static Profile convertProfile(org.wso2.carbon.mdm.beans.Profile mdmProfil
         profile.setDeviceType(mdmProfile.getDeviceType());
 
         List<org.wso2.carbon.policy.mgt.common.ProfileFeature> profileFeatures = new ArrayList<org.wso2.carbon.policy
-                .mgt.common.ProfileFeature>();
+                .mgt.common.ProfileFeature>(mdmProfile.getProfileFeaturesList().size());
         for(ProfileFeature mdmProfileFeature:mdmProfile.getProfileFeaturesList()){
```

Comment:
Recommended fixes  can be found in PR[1]

[1]. https://github.com/wso2/product-mdm/pull/158